internal/pkg/cli/svc_deploy.go

```diff
@@ -591,7 +593,62 @@ func validateAlias(svcName, alias string, app *config.Application, envName strin
 `, color.HighlightCode("http.alias"), envName, app.Name, app.Domain, envName,
 		app.Name, app.Domain, app.Name, app.Domain, app.Name,
 		app.Domain, app.Domain, app.Domain)
-	return fmt.Errorf("alias is not supported in hosted zones not managed by Copilot")
+	return fmt.Errorf("alias is not supported in hosted zones that are not managed by Copilot")
+}
+
+func checkUnsupportedAlias(alias, envName string, app *config.Application) error {
+	var regEnvHostedZone, regAppHostedZone *regexp.Regexp
+	var err error
+	if regEnvHostedZone, err = regexp.Compile(fmt.Sprintf(`^([^\.]+\.)?%s.%s.%s`, envName, app.Name, app.Domain)); err != nil {
```

Comment:
nit: would it be helpful to future code-readers to give an example of valid regex here and/or below?